iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/table/TreeViewSchema.java

```
@@ -0,0 +1,97 @@
+
+  public static boolean isTreeViewTable(final TsTable table) {
+    return table.getPropValue(TREE_PATH_PATTERN).isPresent();
+  }
+
+  public static PartialPath getPrefixPattern(final TsTable table) {
+    return table
+        .getPropValue(TreeViewSchema.TREE_PATH_PATTERN)
+        .map(TreeViewSchema::forceSeparateStringToPartialPath)
+        .orElseThrow(
+            () ->
+                new IoTDBRuntimeException(
+                    String.format(
+                        "Failed to get the original database, because the %s is null for table %s",
+                        TreeViewSchema.TREE_PATH_PATTERN, table.getTableName()),
+                    TSStatusCode.SEMANTIC_ERROR.getStatusCode()));
+  }
+
+  public static PartialPath forceSeparateStringToPartialPath(final String string) {
+    final PartialPath partialPath;
+    try {
+      partialPath = new PartialPath(string);
+    } catch (final IllegalPathException e) {
+      throw new IoTDBRuntimeException(
+          String.format(
+              "Failed to parse the tree view string %s when convert to IDeviceID", string),
+          TSStatusCode.SEMANTIC_ERROR.getStatusCode());
+    }
+    return partialPath;
+  }
+
+  public static String getOriginalName(final TsTableColumnSchema schema) {
+    return schema.getProps().get(ORIGINAL_NAME);
+  }
+
+  public static void setOriginalName(final TsTableColumnSchema schema, final String name) {
+    schema.getProps().put(ORIGINAL_NAME, name);
+  }
+
+  public static boolean isRestrict(final TsTable table) {
+    return table.getPropValue(RESTRICT).isPresent();
+  }
+
+  public static void setRestrict(final TsTable table) {
+    table.addProp(RESTRICT, "");
+  }
+
+  public static String setPathPattern(final TsTable table, final PartialPath pathPattern) {
+    final String[] nodes = pathPattern.getNodes();
+    if (!PathPatternUtil.isMultiLevelMatchWildcard(nodes[nodes.length - 1])) {
+      return "The last node must be '**'";
```

You can throw an exception in this method

---

This class is in "common" and cannot throw the "SemanticException" because the latter one is in "server" module...